frontend/src/tests/wage-chunk-tests.js

```diff
@@ -1,12 +1,221 @@
+            rate: 12.01,
+        };
+
+        const resp = await apiPOST(`/admin/wage_chunks`, updatedWageChunkData);
+        expect(resp).toHaveStatus("success");
+
+        const { payload: updatedWageChunk } = resp;
+        expect(updatedWageChunk).toMatchObject(updatedWageChunkData);
+        checkPropTypes(wageChunkPropTypes, updatedWageChunk);
+        expect(updatedWageChunk.id).not.toBeNull();
+        expect(updatedWageChunk.id).toEqual(originalWageChunk.id);
+
+        // get the wage chunk list and make sure we're updated there as well
+        const { payload: withUpdatedWageChunk } = await apiGET(
+            `/admin/assignments/${assignment.id}/wage_chunks`
+        );
+        expect(withUpdatedWageChunk).toContainObject(updatedWageChunkData);
+        expect(withUpdatedWageChunk.length).toEqual(2);
+    });
+
+    it("delete wage_chunk", async () => {
+        const resp = await apiPOST(
+            `/admin/wage_chunks/delete`,
+            originalWageChunk
+        );
+        expect(resp).toHaveStatus("success");
+
+        // get the wage chunk list and make sure we're updated there as well
+        const { payload: withUpdatedWageChunk } = await apiGET(
+            `/admin/assignments/${assignment.id}/wage_chunks`
+        );
+        expect(withUpdatedWageChunk).not.toContainObject(originalWageChunk);
+        expect(withUpdatedWageChunk.length).toEqual(1);
+    });
+
+    it("automatic creation of wage chunk for position when `hours` is set", async () => {
+        // create a new position to assign
+        const newPositionData = {
+            position_code: "CSC100F",
+            position_title: "Basic Computer Science",
+            hours_per_assignment: 150,
+            start_date: "2019/09/09",
+            end_date: "2019/12/31",
+            contract_template_id: contractTemplate.id,
+        };
+
+        const { payload: position } = await apiPOST(
+            `/admin/sessions/${session.id}/positions`,
+            newPositionData
+        );
+
+        // create new assignment
+        const newAssignmentData = {
+            note: "",
+            hours: 150,
+            position_id: position.id,
+            applicant_id: applicant.id,
+            start_date: "2019-09-02T00:00:00.000Z",
+            end_date: "2019-12-31T00:00:00.000Z",
+        };
+
+        const { payload: newAssignment } = await apiPOST(
+            "/admin/assignments",
+            newAssignmentData
+        );
+
+        const resp = await apiGET(
+            `/admin/assignments/${newAssignment.id}/wage_chunks`
+        );
+        expect(resp).toHaveStatus("success");
+
+        const wageChunkData = {
+            hours: newPositionData.hours_per_assignment,
+        };
+
+        // make sure the position we created is in that list
+        expect(resp.payload).toContainObject(wageChunkData);
+
+        checkPropTypes(PropTypes.arrayOf(wageChunkPropTypes), resp.payload);
+    });
```

Comment:
Is there a test to ensure when setting the wage chunks via an array (on the `assignments` route), that the non-listed wage chunks are deleted?

Reply:
I have done that by:

- (line 111) ensure that the list of new wage chunks contains all wage chunks we have sent to the API
- (line 131) mark sure that only 2 wage chunks are returned.

But I agree that it is not direct enough to show the non-listed wage chunks are no longer there, so I have added line 118 to make sure that non-listed wage chunk's id is not in the `payload`.

Had to change the mock API a little bit to let it pass the test because the real API upserts the listed wage chunks first and then delete the extra ones (so that if a new wage chunk is added, the `id` will be different than the non-listed one) while the mock API delete the extra wage chunks before upserting listed wage chunks. Both way works, but for the sake of testing, I am changing the mock API to match real API because IMO comparing `id` shows more directly that the non-listed one is deleted instead of updated.